src/Control/Controller.php

```diff
@@ -528,9 +533,15 @@ public function render($params = null)
      * Call this to disable site-wide basic authentication for a specific controller. This must be
      * called before Controller::init(). That is, you must call it in your controller's init method
      * before it calls parent::init().
+     *
+     * @deprecated 4.0..5.0
      */
     public function disableBasicAuth()
     {
+        Deprecation::notice(
+            '5.0',
+            'Add this controller\'s url to ' . BasicAuthMiddleware::class . '.URLPatterns property instead'
```

Comment:
does casing matter? I see `urlPatterns` and `URLPatterns` used interchangeably, could we keep it to one case?

Reply:
`URLPatterns` is public setter `$middleware->URLPatterns`. `$urlPatterns` is the protected var. :)

Reply:
oh, you're right :D nice one